go-controller/pkg/libovsdbops/vswitchd.go

```diff
@@ -0,0 +1,279 @@
+	deleted := []*vswitchdb.Interface{}
+	opModel := operationModel{
+		ModelPredicate: p,
+		ExistingResult: &deleted,
+		ErrNotFound:    false,
+		BulkOp:         true,
+	}
+
+	m := newModelClient(vsClient)
+	return m.Delete(opModel)
+}
+
+// FindBridgeByName finds a bridge by name
+func FindBridgeByName(vsClient libovsdbclient.Client, bridgeName string) (*vswitchdb.Bridge, error) {
+	m := newModelClient(vsClient)
+	bridge := &vswitchdb.Bridge{Name: bridgeName}
+	if err := m.Lookup(operationModel{
+		Model:       bridge,
+		ErrNotFound: true,
+	}); err != nil {
+		return nil, err
+	}
+	return bridge, nil
+}
+
+// DeletePorts deletes the given OVS ports by name
+func DeletePort(vsClient libovsdbclient.Client, bridgeName, portName string) error {
+	m := newModelClient(vsClient)
+
+	bridge := &vswitchdb.Bridge{Name: bridgeName}
+	foundPorts := []*vswitchdb.Port{}
+	ops, err := m.LookupOps(nil, operationModel{
+		Model:          &vswitchdb.Port{Name: portName},
+		ExistingResult: &foundPorts,
+		DoAfter: func() {
+			if len(foundPorts) > 0 {
+				bridge.Ports = []string{foundPorts[0].UUID}
+			}
+		},
+		ErrNotFound: false,
+	})
+	if err != nil {
+		return err
+	}
+
+	opModels := []operationModel{
+		{
+			Model:            bridge,
+			OnModelMutations: []interface{}{&bridge.Ports},
+			ErrNotFound:      true,
+			BulkOp:           false,
+		},
+		{
+			Model: &vswitchdb.Interface{Name: portName},
+		},
+		{
+			Model: &vswitchdb.Port{Name: portName},
+		},
+	}
+	ops, err = m.DeleteOps(ops, opModels...)
+	if err != nil {
+		return err
+	}
+
+	_, err = TransactAndCheck(vsClient, ops)
+	return err
+}
+
+// ClearPortQoSBySandboxID clears QoS for any port that has a corresponding interface
+// with the given sandbox ID
+func ClearPortQoSBySandboxID(vsClient libovsdbclient.Client, sandboxID string) error {
+	portNames := []string{}
+	opModels := make([]operationModel, 0, 2)
+	opModels = append(opModels, operationModel{
+		ExistingResult: &[]*vswitchdb.Interface{},
+		ModelPredicate: func(item *vswitchdb.Interface) bool {
+			foundID, ok := item.ExternalIDs["sandbox"]
+			if ok && foundID == sandboxID {
+				portNames = append(portNames, item.Name)
+				return true
+			}
+			return false
+		},
+		ErrNotFound: false,
+		BulkOp:      true,
+	})
+
+	portNoQoS := vswitchdb.Port{}
+	opModels = append(opModels, operationModel{
+		ExistingResult: &[]*vswitchdb.Port{},
+		ModelPredicate: func(item *vswitchdb.Port) bool {
+			for _, name := range portNames {
+				if item.Name == name {
+					return true
+				}
+			}
+			return false
+		},
+		OnModelUpdates: []interface{}{&portNoQoS.QOS},
+		ErrNotFound:    false,
+		BulkOp:         true,
+	})
+
+	m := newModelClient(vsClient)
+	_, err := m.Update(opModels...)
+	return err
+}
+
+// SetInterfaceOVNInstalled sets an OVS Interface's ovn-installed ExternalID.
+// Should only be used from testcases.
+func SetInterfaceOVNInstalled(vsClient libovsdbclient.Client, name string, ovnInstalled bool) error {
```

Comment:
We now have `pkg/testing/libovsdb/ops.go` for these type of testing functions.

Reply:
Moved and reworked to not use libovsdbops internals.